src/index.js

```diff
@@ -12,12 +19,14 @@ class Anvil {
   //   apiKey: <yourAPIKey>,
   //   accessToken: <yourAPIKey>, // OR oauth access token
   //   baseURL: 'https://app.useanvil.com'
+  //   userAgent: 'Anvil API Client/2.0.0'
   // }
   constructor (options) {
     if (!options) throw new Error('options are required')
-    this.options = Object.assign({}, defaultOptions, options)
     if (!options.apiKey && !options.accessToken) throw new Error('apiKey or accessToken required')
 
+    this.options = Object.assign({}, defaultOptions, options)
```

Comment:
for a custom supplied `userAgent` would we still want to tack on the `/<version>`? Would let us know which client version they were using. I suppose this isn't used currently though.

Reply:
We _could_ do that...but it sort of felt to me like if the User was overriding it that we shouldn't tinker with things. We decided to leave this undocumented for now anyways.

Another thing we could consider is always adding a custom header indicating that it came from our node module.

For now, it's nice to get our client in the user agent header though!